ql/src/java/org/apache/hadoop/hive/ql/plan/ExprNodeDescUtils.java

```diff
@@ -1001,17 +1002,10 @@ public static boolean isSame(List<ExprNodeDesc> first, List<ExprNodeDesc> second
   // Given an expression this method figures out if the type for the expression belongs to string group
   // e.g. (String, Char, Varchar etc)
   public static boolean isStringType(ExprNodeDesc expr) {
-    TypeInfo typeInfo = expr.getTypeInfo();
-    if (typeInfo.getCategory() == ObjectInspector.Category.PRIMITIVE) {
-      PrimitiveObjectInspector.PrimitiveCategory primitiveCategory = ((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory();
-      if (PrimitiveObjectInspectorUtils.getPrimitiveGrouping(primitiveCategory) ==
-          PrimitiveObjectInspectorUtils.PrimitiveGrouping.STRING_GROUP) {
-        return true;
-      }
-    }
-    return false;
+    return TypeCheckProcFactory.isStringType(expr.getTypeInfo());
```

Comment:
ExprNodeDescUtils.isStringType is only redirecting to TypeCheckProcFactory.isStringType, therefore it can be removed, and it's callers may call TypeCheckProcFactory.isStringType directly.

Reply:
It was not called anymore, I have just removed it.